### PR TITLE
Improve Daylight request load

### LINKED ISSUE #3087: Switch to on-demand strategy for polling for daylight abilities.
We need to refactor the periodic polling pattern used in [pollForAbilities](https://github.com/tahowallet/extension/blob/main/background/services/abilities/index.ts#L131:L131) to one that requests abilities on-demand when the wallet is opened by the user.

After the refactor - the following should be true:
- The extension does _NOT_ get abilities when it starts up.
- The extension does _NOT_ get abilities every hour.
- The extension _DOES_ get abilities when the extension window is opened.
- Getting abilities should happen at most once per hour.

### PR DESCRIPTION
**Pull request by 0xDaedalus**
Closes #3087 

### Testing
The following should be true in order for this PR to pass testing:

- [x] The extension does NOT get abilities when it starts up.
- [x] The extension does NOT get abilities every hour.
- [x] The extension DOES get abilities when the extension window is opened.
- [x] Getting abilities should happen at most once per hour.
- [x] When importing a new account - that accounts abilities should be fetched on import
- [x] When adding an address to an already imported account - that addresses abilities should be fetched when added.

Latest build: [extension-builds-3095](https://github.com/tahowallet/extension/suites/11302110057/artifacts/580094672) (as of Thu, 02 Mar 2023 09:47:03 GMT).

### CODE REVIEW

**Review by hyphenized — CHANGES_REQUESTED** [review]
Nicely done! Left a couple of comments, aside from that, it's looking great.

**Review by kkosiorowska — CHANGES_REQUESTED** [review]
Looks really good! However, I did find a small thing. When I am trying to add a ledger account, requests are sent for all addresses before I select anyone.

https://user-images.githubusercontent.com/23117945/222096119-189d4911-8be9-41e3-9313-da0c33f72a85.mov

File: background/services/abilities/index.ts [hunk 1]
```diff
@@ -113,26 +105,15 @@ export default class AbilitiesService extends BaseService<Events> {
...
 
-  async pollForAbilities(address: HexString): Promise<void> {
-    if (!isEnabled(FeatureFlags.SUPPORT_ABILITIES)) {
-      return
-    }
+  // Should only be called with ledger or imported accounts
+  async getNewAccountAbilities(address: string): Promise<void> {
+    this.pollForAbilities(address)
+    this.emitter.emit("newAccount", address)
+  }
```
[the remark this answers is not in the document]
Comment (0xDaedalus) [reply background/services/abilities/index.ts:114]: whoopsie - good catch here

File: background/main.ts [hunk 2]
```diff
@@ -932,6 +929,7 @@ export default class Main extends BaseService<never> {
     })
 
     uiSliceEmitter.on("userActivityEncountered", (addressOnNetwork) => {
+      this.abilitiesService.refreshAbilities()
```
[the remark this answers is not in the document]
Comment (0xDaedalus) [reply background/main.ts:932]: Yup - I think this is fine for now tbh.

**Comment by 0xDaedalus** [comment]
Most feedback addressed - won't have time to dig into the multiple ledger requests before the next release so have created a follow-up issue: https://github.com/tahowallet/extension/issues/3100

File: background/services/abilities/index.ts [hunk 3]
```diff
@@ -177,10 +158,13 @@ export default class AbilitiesService extends BaseService<Events> {
     }
   }
 
-  async abilitiesAlarm(): Promise<void> {
-    if (!isEnabled(FeatureFlags.SUPPORT_ABILITIES)) {
+  async refreshAbilities(): Promise<void> {
+    const lastFetchTime = localStorage.getItem(this.ABILITY_TIME_KEY)
+
+    if (lastFetchTime && Number(lastFetchTime) + HOUR < Date.now()) {
```
Comment (kkosiorowska) [inline background/services/abilities/index.ts:164]: It seems that the `<` should be set in reverse. Which makes more sense.  If lastFetchTime + hour is greater than the current time we shouldn't fetch data. 

I'll prepare a fix for this.

https://user-images.githubusercontent.com/23117945/222387316-c3aea1ba-b4f5-4327-a646-79e7f9e3966f.mov

**Comment by kkosiorowska** [comment]
It would be good to have another person look at it after revisions. @jagodarybacka please take a look if everything works as expected.

**Review by jagodarybacka — APPROVED** [review]
✅

**Pull request merged**